src/main/java/org/springframework/data/mapping/callback/EntityCallbackDiscoverer.java

```diff
@@ -394,7 +394,7 @@ private BeanFactory getRequiredBeanFactory() {
 					}
 				}
 
-				return cachedEntityCallbacks;
+				return new ArrayList<>(cachedEntityCallbacks);
```

Comment:
I fixed it again.

I think it's safer to handle it where it returns the result.